controller/konnect/watch_kongpluginbinding.go

```diff
@@ -306,3 +336,135 @@ func enqueueKongPluginBindingForKongClusterPlugin(cl client.Client) func(
 		})
 	}
 }
+
+func enqueueKongPluginBindingForKongService(cl client.Client) func(
+	ctx context.Context, obj client.Object) []reconcile.Request {
+	return func(ctx context.Context, obj client.Object) []reconcile.Request {
+		kongService, ok := obj.(*configurationv1alpha1.KongService)
+		if !ok {
+			return nil
+		}
+
+		pluginBindingList := configurationv1alpha1.KongPluginBindingList{}
+		err := cl.List(ctx, &pluginBindingList,
```

Comment:
Ditto here. Since the reference to `KongService`/`KongRoute`/`KongConusmer`/`KongConusmerGroup` are limited in the same namespace, I think we could add a `InNamespace` constraint for the listing.